app/models/miq_worker.rb

```diff
@@ -154,7 +154,7 @@ def self.sync_workers
         (desired - current).times { result[:adds] << self.start_worker.pid }
       elsif desired < current
         (current - desired).times do
-          ww = w.pop
+          ww = w.to_a.pop
```

Comment:
That sounds unlikely... if we're popping, we surely want to use the same array throughout the loop.